pkg/comp-functions/functions/vshnpostgres/postgresql_deploy.go

```diff
@@ -2,31 +2,604 @@ package vshnpostgres
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Errorf("cannot create podMonitor object: %w", err).Error())
+	}
+
+	if comp.Spec.Parameters.Restore != nil {
+		l.Info("Create copy job")
+		err = createCopyJob(comp, svc)
+		if err != nil {
+			return runtime.NewWarningResult(fmt.Errorf("cannot create copyJob object: %w", err).Error())
+		}
+	}
+	return nil
+}
+
+func createCerts(comp *vshnv1.VSHNPostgreSQL, svc *runtime.ServiceRuntime) error {
+	selfSignedIssuer := &cmv1.Issuer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName(),
+			Namespace: comp.GetInstanceNamespace(),
+		},
+		Spec: cmv1.IssuerSpec{
+			IssuerConfig: cmv1.IssuerConfig{
+				SelfSigned: &cmv1.SelfSignedIssuer{
+					CRLDistributionPoints: []string{},
+				},
+			},
+		},
+	}
+
+	err := svc.SetDesiredKubeObjectWithName(selfSignedIssuer, comp.GetName()+"-localca", "local-ca")
+	if err != nil {
+		err = fmt.Errorf("cannot create local ca object: %w", err)
+		return err
+	}
+
+	certificate := &cmv1.Certificate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName(),
+			Namespace: comp.GetInstanceNamespace(),
+		},
+		Spec: cmv1.CertificateSpec{
+			SecretName: certificateSecretName,
+			Duration: &metav1.Duration{
+				Duration: time.Duration(87600 * time.Hour),
+			},
+			RenewBefore: &metav1.Duration{
+				Duration: time.Duration(2400 * time.Hour),
+			},
+			Subject: &cmv1.X509Subject{
+				Organizations: []string{
+					"vshn-appcat",
+				},
+			},
+			IsCA: false,
+			PrivateKey: &cmv1.CertificatePrivateKey{
+				Algorithm: cmv1.RSAKeyAlgorithm,
+				Encoding:  cmv1.PKCS1,
+				Size:      4096,
+			},
+			Usages: []cmv1.KeyUsage{"server auth", "client auth"},
+			DNSNames: []string{
+				comp.GetName() + "." + comp.GetInstanceNamespace() + ".svc.cluster.local",
+				comp.GetName() + "." + comp.GetInstanceNamespace() + ".svc",
+			},
+			IssuerRef: certmgrv1.ObjectReference{
+				Name:  comp.GetName(),
+				Kind:  "Issuer",
+				Group: "cert-manager.io",
+			},
+		},
+	}
+
+	err = svc.SetDesiredKubeObjectWithName(certificate, comp.GetName()+"-certificate", "certificate")
+	if err != nil {
+		err = fmt.Errorf("cannot create local ca object: %w", err)
+		return err
 	}
 
 	return nil
+}
+
+func createStackgresObjects(ctx context.Context, comp *vshnv1.VSHNPostgreSQL, svc *runtime.ServiceRuntime) error {
+
+	err := createSgInstanceProfile(ctx, comp, svc)
+
+	if err != nil {
+		return err
+	}
+
+	err = createSgPostgresConfig(comp, svc)
 
+	if err != nil {
+		return err
+	}
+
+	err = createSgCluster(ctx, comp, svc)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createSgInstanceProfile(ctx context.Context, comp *vshnv1.VSHNPostgreSQL, svc *runtime.ServiceRuntime) error {
+	plan := comp.Spec.Parameters.Size.GetPlan(svc.Config.Data["defaultPlan"])
+
+	resources, err := utils.FetchPlansFromConfig(ctx, svc, plan)
+	if err != nil {
+		err = fmt.Errorf("cannot fetch plans from the composition config, maybe they are not set: %w", err)
```

Comment:
Don't assign `err` just to return it. Return the `fmt.Errorf()` directly.

Reply:
As I'm also using `svc.AddResult()` now, I still need the `err` variable.